### PR TITLE
preselections metadata

### PR DESCRIPTION
**Pull request by wschidol**
The next version of ISO/IEC 14496-12 is poised to add structures to represent preselections through codec-agnostic metadata. This PR adds support for the meta, grpl, prsl, ardi, elng, and labl boxes. Along the way, a new datatype utf8string is provided to be able to correctly interpret UTF8 strings. The boxes have unit tests provided against them.

This PR implements the syntax defined in the FDIS version of ISO/IEC 14496-12 and obsoletes #57 .

### CODE REVIEW

**Comment by wschidol** [comment]
"press release" for FDIS: [https://dms.mpeg.expert/news/news.php?id=80](https://dms.mpeg.expert/news/news.php?id=80)

**Comment by wschidol** [comment]
@dsilhavy @bbert let me know if there is anything I can help to push this one along.

File: src/processors/prsl.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// ISO/IEC 14496-12:202x - 8.18.4.1 Preselection group box
+ISOBox.prototype._boxProcessors['prsl'] = function() {
+  this._procFullBox();
+  this._procField('group_id', 'uint', 32);
+  this._procField('num_entities_in_group', 'uint', 32);
+  this._procEntries('entities', this.num_entities_in_group, function(entry) {
+    this._procEntryField(entry, 'entity_id', 'uint', 32);
+  });
+  if (this.flags & 0x1000) this._procField('preselection_tag', 'utf8string');
```
Comment (dsilhavy) [inline src/processors/prsl.js:9]: I couldn't find the definition of the flags in the document you provided, but I assume they are aligned with what is in the spec

**Review by dsilhavy — APPROVED** [review]

**Comment by dsilhavy** [comment]
@wschidol This looks good to me, thanks for providing the changes. 

@bbert In my opinion this is good to go, and we can release v.0.3.10.

**Pull request merged**